src/features/search/components/locationChoice.utils.tsx

```diff
@@ -0,0 +1,33 @@
+import { t } from '@lingui/macro'
+import React from 'react'
+
+import { LocationChoiceType } from 'features/search/locationChoice.types'
+import { _ } from 'libs/i18n'
+import { AroundMe } from 'ui/svg/icons/AroundMe'
+import { Everywhere } from 'ui/svg/icons/Everywhere'
+import { ColorsEnum } from 'ui/theme'
+
+export const getLocationChoiceName = (locationChoice: LocationChoiceType): string => {
+  switch (locationChoice) {
+    case LocationChoiceType.LOCALIZED:
+      return _(t`Autour de moi`)
+    case LocationChoiceType.EVERYWHERE:
+      return _(t`Partout`)
+    default:
+      return _(t`Partout`)
+  }
+}
+
+export const getLocationChoiceIcon = (
+  locationChoice: LocationChoiceType,
+  selected: boolean
+): JSX.Element => {
+  switch (locationChoice) {
+    case LocationChoiceType.LOCALIZED:
+      return <AroundMe size={48} color2={selected ? ColorsEnum.PRIMARY : ColorsEnum.SECONDARY} />
```

Comment:
```suggestion
      return AroundMe
```

vu que toutes les props sont les mêmes

Reply:
il y a la logique pour la color2 quand meme non ?

Reply:
si tu changes cette fonction en:

```javascript
export const getLocationChoiceIcon = (
  locationChoice: LocationChoiceType
): React.FC<BicolorIconInterface> => {
  switch (locationChoice) {
    case LocationChoiceType.LOCALIZED:
      return AroundMe
    case LocationChoiceType.EVERYWHERE:
      return Everywhere
    default:
      return AroundMe
  }
}
```

tu peux ensuite l'utiliser comme ça:

````javascript
  const iconColor2 = isSelected ? ColorsEnum.PRIMARY : ColorsEnum.SECONDARY
  const LocationChoiceIcon = getLocationChoiceIcon(type)

  return (
    <Container onPress={() => setIsSelected(!isSelected)}>
      <FirstPart>
        <LocationChoiceIcon size={48} color2={iconColor2} />
```

ce qui permet de factoriser size et selected

Reply:
ah oui ! je fais ça